thoth/user_api/api_v1.py

```diff
@@ -590,22 +607,27 @@ def list_python_packages(
     query_result = GRAPH.get_python_package_version_names_all(
         sort=True,
         distinct=True,
-        start_offset=page,
+        start_offset=page * PAGINATION_SIZE,
+        count=PAGINATION_SIZE,
         os_name=os_name,
         os_version=os_version,
         python_version=python_version,
     )
 
-    return {"packages": [{"package_name": i} for i in query_result]}, 200
+    return (
+        {"packages": [{"package_name": i} for i in query_result]},
+        200,
+        {"page": page, "per_page": PAGINATION_SIZE, "page_count": None},  # TODO: page count
+    )
```

Comment:
CC @Gkrumbach07 PTAL if this needs any adjustments. Each request to an endpoint that is paginated will return the following information in HTTP headers with this change:

* `page` - the current page
* `per_page` - page size
* `page_count` - number of pages total

Reply:
Page size could be redundant given that the list of elements on that page is given. However that is not an issue, so LGTM.

Reply:
IMHO, it might be a good idea to keep page size - it is a backend configuration that can change. Also, if the response has, let's say, 3 entries, it is not clear what the page size is (assuming the page size is more than 3).

Reply:
One possible feature I see missing here, that is available in other paginated apis, is a limit option. This messes up the paging I assume so adjusting the parameters may be needed. This new solution uses standard naming and allows to have a custom page size. This is helpful in cases where you might not want to display 100 elements all at once. Let me know what you think about this. 

For example for request:
`limit`: how many elements to return (possibly allow negative indexes to start from the back)
`start`: what index to start at

For response:
`limit`: equal to the request param
`size`: number of elements _returned_
`next`: the next index of the entire list (null if at the end)
`prev`: the index to get the the last page (`start` - `limit` or 0)

Reply:
OK, sounds good - I tried to follow [GitHub API pagination design](https://docs.github.com/en/rest/guides/traversing-with-pagination). Similarly, we can have page size configurable, `prev` and `next` in that case. I'll have a look 👍🏻 

Would negative indexing support have any advantages on UI? I'm wondering if we need to bring it.

Reply:
I've implemented also negative indexing. I'm not quite sure how to properly behave in certain situations to make your work easier, so eager to have your feedback. Basically, negative indexing cycles but positive indexing stops at the max page.

Reply:
Awesome that will be helpful, the cycling shouldn't be an issue, as I can just do some arithmetic to figure out if there is any entries left.